backend/entityservice/tasks/comparing.py

```diff
@@ -221,34 +283,45 @@ def aggregate_comparisons(similarity_result_files, project_id, run_id, parent_sp
     log = logger.bind(pid=project_id, run_id=run_id)
     if similarity_result_files is None:
         raise TypeError("Inappropriate argument type - missing results files.")
-    mc = connect_to_object_store()
-    files = []
-    data_size = 0
 
+    files = []
     for res in similarity_result_files:
         if res is None:
             log.warning("Missing results during aggregation. Stopping processing.")
             raise TypeError("Inappropriate argument type - results missing at aggregation step.")
+        num, filesize, filename = res
+        if num:
+            assert filesize is not None
+            assert filename is not None
+            files.append(res)
         else:
-            num, filename = res
+            assert filesize is None
+            assert filename is None
+    heapq.heapify(files)
 
-        if num > 0:
-            files.append(filename)
-            data_size += mc.stat_object(Config.MINIO_BUCKET, filename).size
+    log.debug(f"Aggregating result chunks from {len(files)} files, "
+              f"total size: {sum(map(operator.itemgetter(1), files))}")
 
-    log.debug("Aggregating result chunks from {} files, total size: {}".format(
-        len(files), fmt_bytes(data_size)))
+    mc = connect_to_object_store()
+    while len(files) > 1:
+        file0 = heapq.heappop(files)
+        file1 = heapq.heappop(files)
+        merged_file = _merge_files(mc, log, file0, file1)
+        heapq.heappush(files, merged_file)
```

Comment:
Perhaps delete file0 and file1 at this point.

Reply:
They are deleted on [line 257](#diff-30fb0765378c9296147bce2d345daed9R257).